### ISSUE TITLE
No highlight cancels out cropping

### ISSUE THREAD

**Issue by bidoubiwa** [meilisearch/MeiliSearch#449]
When adding `attributesToHighlight` to my search query, if no matched query word can be highlighted it will also cancel out the cropping.

```
      q: 'scientist',
      offset: 0,
      limit: 5,
      attributesToRetrieve: ['overview', 'id'],
      attributesToSearchIn: ['overview'],
      attributesToCrop: ['overview'],
      cropLength: 1,
      attributesToHighlight: ['overview']
```
Response : 
```
  res: {
    id: '485811',
    overview: 'After a zombie apocalypse spreads from a London prison, the UK is brought to its knees. The spread of the virus is temporarily contained but, without a cure, it’s only a matter of time before it breaks its boundaries and the biggest problem of all… any zombies with combat skills are now enhanced. With the South East of England quarantined from the rest of the world using fortified borders, intelligence finds that the scientist responsible for the outbreak is alive and well in London. With his recovery being the only hope of a cure, a squad of eight Special Forces soldiers is sent on a suicide mission to the city, now ruled by the undead, with a single task: get him out alive within 72 hours by any means necessary. What emerges is an unlikely pairing on a course to save humanity against ever-rising odds.'
  },
```
There is no `_formatted` key.

Where there should be this : 
```
  res: {
    id: '485811',
    overview: 'After a zombie apocalypse spreads from a London prison, the UK is brought to its knees. The spread of the virus is temporarily contained but, without a cure, it’s only a matter of time before it breaks its boundaries and the biggest problem of all… any zombies with combat skills are now enhanced. With the South East of England quarantined from the rest of the world using fortified borders, intelligence finds that the scientist responsible for the outbreak is alive and well in London. With his recovery being the only hope of a cure, a squad of eight Special Forces soldiers is sent on a suicide mission to the city, now ruled by the undead, with a single task: get him out alive within 72 hours by any means necessary. What emerges is an unlikely pairing on a course to save humanity against ever-rising odds.',
    _formatted: { id: '485811', overview: ' s' }
  },
```

**Comment by bidoubiwa** [comment]
**This will be fixed in v0.9.0.**

**Comment by qdequele** [comment]
closed by #441
